apps/web/src/app/student/quiz/_components/QuizView.tsx

```diff
@@ -0,0 +1,107 @@
+"use client";
+
+import { ButtonGroup, ButtonGroupItem } from "src/components/ui/ButtonGroup";
+import { trpc } from "../../../../utils/trpc/client";
+import { TrpcReactQueryOptions } from "../../../../utils/trpc/lib";
+import { Button } from "@/components/ui/button";
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import * as z from "zod";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormMessage
+} from "@/components/ui/form";
+import { Heading } from "src/components/ui";
+import { CheckCircledIcon } from "@radix-ui/react-icons";
+import { useState } from "react";
+import { Progress } from "@/components/ui/progress";
+
+interface Props {
+  id: number;
+  initialData: TrpcReactQueryOptions["questionBank"]["get"]["initialData"];
+}
+export const QuizView = ({ id, initialData }: Props) => {
+  const { data: questionBank, refetch } = trpc.questionBank.get.useQuery(id, {
+    initialData
+  });
+  const [isSubmitted, setIsSubmitted] = useState<boolean>(false);
+  const question_id = 0; //mock question ID
+  const formSchema = z.object({
+    answer_id: z.coerce.number({
+      required_error: "Please select an answer",
+      invalid_type_error: "Please select an answer"
+    })
+  });
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema)
+  });
+  if (!questionBank) {
+    return <p>Question bank doesnt exist</p>;
```

Comment:
student shouldn't see this error. perhaps generic quiz not found?

Reply:
Changed in https://github.com/peiyuans-pals/toohakai/pull/36/commits/6479c6e4f9ade3c9fd4341c4ea9fa240159108db
```html
<div className="flex items-center justify-center h-screen">Quiz not found.</div>
```

Reply:
nice